In [1]:
# Read in the CSV file
import csv 
f = open("guns.csv","r")
data = list(csv.reader(f))

print(data[:5])

[['', 'year', 'month', 'intent', 'police', 'sex', 'age', 'race', 'hispanic', 'place', 'education'], ['1', '2012', '01', 'Suicide', '0', 'M', '34', 'Asian/Pacific Islander', '100', 'Home', '4'], ['2', '2012', '01', 'Suicide', '0', 'F', '21', 'White', '100', 'Street', '3'], ['3', '2012', '01', 'Suicide', '0', 'M', '60', 'White', '100', 'Other specified', '4'], ['4', '2012', '02', 'Suicide', '0', 'M', '64', 'White', '100', 'Home', '4']]


In [2]:
# Create a class for removing header rows
class Dataset():
    def __init__(self,data):
        self.headers = data[0]
        self.data = data[1:]

In [3]:
a = Dataset(data)
headers = a.headers
data = a.data
print(headers)
print(data[0:5])

['', 'year', 'month', 'intent', 'police', 'sex', 'age', 'race', 'hispanic', 'place', 'education']
[['1', '2012', '01', 'Suicide', '0', 'M', '34', 'Asian/Pacific Islander', '100', 'Home', '4'], ['2', '2012', '01', 'Suicide', '0', 'F', '21', 'White', '100', 'Street', '3'], ['3', '2012', '01', 'Suicide', '0', 'M', '60', 'White', '100', 'Other specified', '4'], ['4', '2012', '02', 'Suicide', '0', 'M', '64', 'White', '100', 'Home', '4'], ['5', '2012', '02', 'Suicide', '0', 'M', '31', 'White', '100', 'Other specified', '2']]


In [4]:
# Count how many gun deaths occur in each year
years = [each[1] for each in data]
year_counts = {}
for each in years:
    if each in year_counts:
        year_counts[each] += 1
    else:
        year_counts[each] = 1
print(year_counts)

{'2014': 33599, '2012': 33563, '2013': 33636}


In [5]:
# Count gun deaths by date
import datetime
dates = [datetime.datetime(year = int(each[1]),
                           month = int(each[2]), 
                           day = 1) for each in data]
dates[:5]

[datetime.datetime(2012, 1, 1, 0, 0),
 datetime.datetime(2012, 1, 1, 0, 0),
 datetime.datetime(2012, 1, 1, 0, 0),
 datetime.datetime(2012, 2, 1, 0, 0),
 datetime.datetime(2012, 2, 1, 0, 0)]

In [6]:
date_counts = {}
for each in dates:
    if each not in date_counts:
        date_counts[each] = 0
    date_counts[each] += 1
date_counts

{datetime.datetime(2012, 1, 1, 0, 0): 2758,
 datetime.datetime(2012, 2, 1, 0, 0): 2357,
 datetime.datetime(2012, 3, 1, 0, 0): 2743,
 datetime.datetime(2012, 4, 1, 0, 0): 2795,
 datetime.datetime(2012, 5, 1, 0, 0): 2999,
 datetime.datetime(2012, 6, 1, 0, 0): 2826,
 datetime.datetime(2012, 7, 1, 0, 0): 3026,
 datetime.datetime(2012, 8, 1, 0, 0): 2954,
 datetime.datetime(2012, 9, 1, 0, 0): 2852,
 datetime.datetime(2012, 10, 1, 0, 0): 2733,
 datetime.datetime(2012, 11, 1, 0, 0): 2729,
 datetime.datetime(2012, 12, 1, 0, 0): 2791,
 datetime.datetime(2013, 1, 1, 0, 0): 2864,
 datetime.datetime(2013, 2, 1, 0, 0): 2375,
 datetime.datetime(2013, 3, 1, 0, 0): 2862,
 datetime.datetime(2013, 4, 1, 0, 0): 2798,
 datetime.datetime(2013, 5, 1, 0, 0): 2806,
 datetime.datetime(2013, 6, 1, 0, 0): 2920,
 datetime.datetime(2013, 7, 1, 0, 0): 3079,
 datetime.datetime(2013, 8, 1, 0, 0): 2859,
 datetime.datetime(2013, 9, 1, 0, 0): 2742,
 datetime.datetime(2013, 10, 1, 0, 0): 2808,
 datetime.datetime(2013, 11,

In [18]:
# Count gun deaths by sex and race
sex = [each[5] for each in data]
sex_counts = {}
for each in sex:
    if each not in sex_counts:
        sex_counts[each] = 0
    sex_counts[each] += 1
    
race = [each[7] for each in data]

race_counts = {}
for each in race:
    if each not in race_counts:
        race_counts[each] = 0
    race_counts[each] += 1
race_counts
sex_counts

{'F': 14449, 'M': 86349}

By exploring `sex` and `race` columns, we can see that male has a significantly higher number of gun deaths than the females (86349 for Male and 14449 for Female). White race is the highest in gun death while Native American/Native Alaskan remains the lowest. I think it would be also helpful to check if education level affect the possibility of gun death happening, which intuitively is a driving factor.

In [19]:
# Read in census data
f = open("census.csv","r")
census = list(csv.reader(f))
print(census)

[['Id', 'Year', 'Id', 'Sex', 'Id', 'Hispanic Origin', 'Id', 'Id2', 'Geography', 'Total', 'Race Alone - White', 'Race Alone - Hispanic', 'Race Alone - Black or African American', 'Race Alone - American Indian and Alaska Native', 'Race Alone - Asian', 'Race Alone - Native Hawaiian and Other Pacific Islander', 'Two or More Races'], ['cen42010', 'April 1, 2010 Census', 'totsex', 'Both Sexes', 'tothisp', 'Total', '0100000US', '', 'United States', '308745538', '197318956', '44618105', '40250635', '3739506', '15159516', '674625', '6984195']]


[['Id',
  'Year',
  'Id',
  'Sex',
  'Id',
  'Hispanic Origin',
  'Id',
  'Id2',
  'Geography',
  'Total',
  'Race Alone - White',
  'Race Alone - Hispanic',
  'Race Alone - Black or African American',
  'Race Alone - American Indian and Alaska Native',
  'Race Alone - Asian',
  'Race Alone - Native Hawaiian and Other Pacific Islander',
  'Two or More Races'],
 ['cen42010',
  'April 1, 2010 Census',
  'totsex',
  'Both Sexes',
  'tothisp',
  'Total',
  '0100000US',
  '',
  'United States',
  '308745538',
  '197318956',
  '44618105',
  '40250635',
  '3739506',
  '15159516',
  '674625',
  '6984195']]

In [9]:
# Counts the portions of death for each races
mapping = {
    "Asian/Pacific Islander": 15159516 + 674625,
    "Native American/Native Alaskan": 3739506,
    "Black": 40250635,
    "Hispanic": 44618105,
    "White": 197318956
}

race_per_hundredk = {}
for k,v in race_counts.items():
    race_per_hundredk[k] = (v/mapping[k]) * 10000

race_per_hundredk

{'Asian/Pacific Islander': 0.8374309664161763,
 'Black': 5.78773477735196,
 'Hispanic': 2.022049121091091,
 'Native American/Native Alaskan': 2.452195557381109,
 'White': 3.356849303419181}

In [20]:
# Explore homicide race counts by race
intents = [each[3] for each in data]
races = [each[7] for each in data]
homicide_race_counts = {}
for i, race in enumerate(races):
    if race not in homicide_race_counts:
        homicide_race_counts[race] = 0
    if intents[i] == "Homicide":
        homicide_race_counts[race] += 1

for k,v in homicide_race_counts.items():
    homicide_race_counts[k] = (v/mapping[k]) * 10000

homicide_race_counts

{'Asian/Pacific Islander': 0.3530346230970155,
 'Black': 4.847128498718095,
 'Hispanic': 1.2627161104219913,
 'Native American/Native Alaskan': 0.8717729026240364,
 'White': 0.46356417981453335}

Black race has the highest proportion of death caused by gun.

In [23]:
# Explore the homicide count by gender
sex = [each[5] for each in data]
homicide_sex_counts = {}
for i,gender in enumerate(sex):
    if gender not in homicide_sex_counts:
        homicide_sex_counts[gender] = 0
    if intents[i] == "Homicide":
        homicide_sex_counts[gender] += 1

homicide_sex_counts

{'F': 5373, 'M': 29803}

In [25]:
# Explore the Accidental, Male count by race
accidental_sex_race_counts = {}
for i,race in enumerate(races):
    if race not in accidental_sex_race_counts:
        accidental_sex_race_counts[race] = 0
    if intents[i] == "Accidental" and sex[i] == "M":
        accidental_sex_race_counts[race] += 1

accidental_sex_race_counts

{'Asian/Pacific Islander': 9,
 'Black': 294,
 'Hispanic': 130,
 'Native American/Native Alaskan': 20,
 'White': 968}